src/controllers/TalksController.php

```diff
@@ -140,8 +142,229 @@ public function postAction($request, $db) {
                     throw new Exception("Operation not supported, sorry", 404);
             }
         } else {
-            throw new Exception("method not yet supported - sorry");
+
+            $event_id = filter_var($request->getParameter('event_id'), FILTER_SANITIZE_NUMBER_INT);
+            $event_mapper = new EventMapper($db, $request);
+            if (! $event_mapper->thisUserHasAdminOn($event_id)) {
+                throw new Exception('You are not an host for this event', 403);
+            }
+
+            // incoming data
+            $talk   = array();
+            $errors = array();
+
+            $talk['event_id'] = $event_id;
+            if (empty($talk['event_id'])) {
+                $errors[] = '"event_id" is a required field';
+            }
+
+            $talk['talk_title'] = filter_var($request->getParameter("talk_title"), FILTER_SANITIZE_STRING);
+            if(empty($talk['talk_title'])) {
+                $errors[] = "'talk_title' is a required field";
+            }
+
+            $talk['url_friendly_talk_title'] = filter_var($request->getParameter("ur_friendly_talk_title"), FILTER_SANITIZE_STRING);
+            if(empty($talk['url_friendly_talk_title'])) {
+                $talk['url_friendly_talk_title'] = $talk['talk_title'];
+            }
+
+            $talk['talk_description']  = filter_var($request->getParameter("talk_description"), FILTER_SANITIZE_STRING);
+            if (empty($talk['talk_description'])) {
+                $errors[] = "'talk_description' is a required field";
+            }
+
+            $talk['type']  = filter_var($request->getParameter("type"), FILTER_SANITIZE_STRING);
+            if (empty($talk['type'])) {
+                $errors[] = "'type' is a required field";
+            }
+            // TODO @heiglandreas check whether the type already exists in the database.
+
+            if ($errors) {
+                throw new Exception(implode(". ", $errors), 400);
+            }
+
+
+            $talk['start_date'] = (new \DateTime($request->getParameter("start_date")))->format('U');
+
+            $talk['duration'] = filter_var(
+                $request->getParameter('duration'),
+                FILTER_SANITIZE_NUMBER_INT
+            );
+            if (empty($talk['duration'])) {
+                $talk['duration'] = 60;
+            }
+
+            $talk['language'] = filter_var($request->getParameter('language'), FILTER_SANITIZE_STRING);
+            if (empty($talk['language'])) {
+                $talk['language'] = 'English - UK';
+            }
+            // TODO: @heiglandreas Check whether the language actually exists in the database
+            // When the language doesn't exist, the talk will not be found
+
+            $talk['slides_link'] = filter_var($request->getParameter('slides_link'), FILTER_SANITIZE_URL);
+
+            $incoming_speakers_list = $request->getParameter('speakers');
```

Comment:
I would have expected speakers to be a sub resource of the talk, though I'll defer to @lornajane's thoughts on this.